src/Block.cpp

```diff
@@ -298,8 +338,14 @@ Block::Output(std::ostream &out, FactMgr* fm, int indent) const
 	if (CGOptions::math_notmp())
 		OutputTmpVariableList(out, indent);
 
+
 	OutputVariableList(local_vars, out, indent);
-	OutputStatementList(stms, out, fm, indent);
+
+	if(CGOptions::check_local()){
+		OutputStatementListCheckLocal(stms, local_vars, out, fm, indent);
```

Comment:
check_local outputs the values of local variables created in each block at the end of it. It may be useful in other applications.